redaxo/src/addons/mediapool/functions/function_rex_mediapool.php

```diff
@@ -94,9 +94,37 @@
     if ($isFileUpload) { // Fileupload?
         $FILETYPE = rex_file::mimeType($FILE['tmp_name']);
 
-        if (!@move_uploaded_file($FILE['tmp_name'], $dstFile)) {
-            $message[] = rex_i18n::msg('pool_file_movefailed');
+        // Bevor die Datei engueltig in den Medienpool uebernommen wird, koennen
+        // Addons ueber einen Extension-Point ein Veto einlegen.
+        // Sobald ein Addon eine negative Entscheidung getroffen hat, sollten
+        // Addons, fuer die der Extension-Point spaeter ausgefuehrt wird, diese
+        // Entscheidung respektieren
+
+        $upload_ep_params = [
+            'file' => $FILE,            // die Datei, die hochgeladen werden soll
+            'file_infos' => $FILEINFOS, // Medienpool-spezifische Infos zur Datei (Name, ...)
+            'user' => $userlogin        // User, der den Upload ausgeloest hat
+        ];
+        $upload_result = [
+            'status' => 'ok',   // 'ok' oder 'declined'
```

Comment:
Vllt MEDIA_BEFORE_UPLOAD ?

Sehen wir weitere use-cases für den EP? Falls ja, ggf. Das handling hier nicht so stark auf den upload-cancel use-case auslegen (oder alternativ einen weniger generischen EP Namen)?

Reply:
Genau genommen ist es ja nicht vor dem Upload, sondern vor dem Verschieben/Finalisieren des bereits geschehenen Uploads.
Ich finde `MEDIA_UPLOAD` als Namen eigentlich ganz ok.

Reply:
Tatsächlich habe ich auch darüber nachgedacht, ob man noch als dritte Möglichkeit einen `pending`-Zustand einführen könnte. Ein Upload könnte auf die Freigabe durch einen Admin warten - bspw. weil es eine große PDF ist, oder ein komisches Dateiformat hochgeladen wurde (warum lädt jemand ein Shellscript hoch?). Wie das genau aussieht, wäre dann Sache des Addons, es bekommt ja einen Verweis auf die Datei mit sämtlichen anderen Informationen übergeben.

Ich bin aus zwei Gründen wieder davon abgerückt:
1. hat der Mediapool soweit ich weiß aktuell keine öffentliche API. Um einen Upload später fortzusetzen (weil der Admin die Datei freigegeben hat) wäre aber genau das notwendig.
2. und fast noch schwerwiegender: es ist für mich unklar, wie sich das System verhalten soll, wenn ein Addon den Upload als `pending` einstuft, ein anderes ihn aber komplett ablehnt. Hier bräuchte man dann irgendeinen Kommunikations- bzw. Verhandlungsmechnismus zwischen den beiden Addons. Damit wird das Protokoll aber sofort ziemlich kompliziert.

Davon abgesehen ist mir nichts eingefallen, wozu man den `MEDIA_UPLOAD` EP bräuchte (den Namen kann ich gerne noch anpassen, `MEDIA_BEFORE_UPLOAD` macht noch klarer, worum es eigentlich geht) und was man insbesondere nicht schon mit `MEDIA_ADDED` machen kann.

Reply:
> Genau genommen ist es ja nicht vor dem Upload, sondern vor dem Verschieben/Finalisieren des bereits geschehenen Uploads.

Wäre dann nicht `MEDIA_UPLOADED` sinnvoller?

Reply:
> Wäre dann nicht `MEDIA_UPLOADED` sinnvoller?

Ganz streng genommen vielleicht ja, ich fände es trotzdem verwirrender.
MEDIA_UPLOAD drückt für mich gut aus, dass man in den Upload-Prozess damit eingreifen kann. Also nach technischem Upload und vor Abschließen des Uploadvorgangs durch Verschieben der Datei zum endgültigen Ort.

Reply:
Alternativ: `MEDIA_ACCEPT` o.ä.?

Reply:
Hm, ich habe gerade Angst, dass sich diese Diskussion verfährt. Einen perfekten Namen zu finden ist vermutlich unmöglich. Wenn es gegen `MEDIA_UPLOAD` also keine großen Einwände gibt - so interpretiere ich die bisherigen Reaktionen - würde ich das erstmal so lassen.

Reply:
MEDIA_PRESAVE ? (In Anlehnung an presave der Modul-Aktionen)